Isodose/qSlicerIsodoseModuleWidget.cxx

```diff
@@ -922,6 +924,29 @@ void qSlicerIsodoseModuleWidget::updateButtonsState()
   d->pushButton_Apply->setEnabled(applyEnabled);
 }
 
+//-----------------------------------------------------------------------------
+void qSlicerIsodoseModuleWidget::onIsodoseRepresentationToggled(bool toggled)
+{
+  Q_D(qSlicerIsodoseModuleWidget);
+
+  vtkMRMLIsodoseNode* paramNode = vtkMRMLIsodoseNode::SafeDownCast(d->MRMLNodeComboBox_ParameterSet->currentNode());
+  if (paramNode)
+  {
+    paramNode->SetSegmentationFlag(toggled);
+  }
+
+  if (!toggled)
+  {
+    d->toggleButton_IsodoseRepresentation->setIcon(QIcon(":/Icons/Isodose.png"));
+    d->toggleButton_IsodoseRepresentation->setToolTip(tr("Represent isodoses as multiple model nodes into folder"));
+  }
+  else
+  {
+    d->toggleButton_IsodoseRepresentation->setIcon(QIcon(":/Icons/Segmentation.png"));

```

Comment:
Why do we have a button for this? I need to know more about this whole work, please give us an overview, and rather write more than less. Thanks.

Reply:
It's a toggle button, if the button is pressed, then segmentation node would be used to store isodose surfaces. If the button isn't pressed, then original isodose representation would be used (model nodes into folder).

Toggle button is pressed in the red square.
![image](https://user-images.githubusercontent.com/3785912/124503307-f433c600-ddcd-11eb-8b2c-ed5a59c9b25f.png)


Isodose surfaces are stored as a segmentation node named "RTDOSE[1]_IsodoseSurfaces"
![image](https://user-images.githubusercontent.com/3785912/124503363-1594b200-ddce-11eb-910b-b82f5cfcd1ac.png)

Reply:
This I understood of course. Why do you need to be able to switch between the two? Why expose it to the user? What is the reason you needed the segmentation representation in the first place?

Reply:
I think it would be enough to only use segmentation. It takes two clicks in Data module to export model hierarchy from a segmentation node.

This work is the result of this discussion: https://github.com/SlicerRt/SlicerRT/issues/80. @MichaelColonel add #80 to the commit comment to make it clear what issue this development resolves. I like this development, because it reduces the complexity of the scene (less nodes) and the visualization is nicer (filled color region and not just the contour).

It might be useful to have two modes:
- A. each segment contains a solid region of dose > threshold
- B. each segment contains  "ring" shaped region of thresholdA < dose < thresholdB

Reply:
I completely agree with these. I was mostly curious why do does this branch introduce this complexity of being able to switch from the module. Switching to segmentation representation instead of model nodes makes sense, but there was nothing in this PR referring to its motivation.

So I'd suggest removing the button and the logic for the switching. It adds unnecessary complexity as much on the UI as in the source code (making it more error-prone). Unless you have a good reason offering this option. Thanks!

Reply:
I will do as you suggested. At the beginning nobody told me to delete initial version of isodose representation, that is why i made a toggle button.

Reply:
I added better description into commit, and implementation of two modes: Single border mode (A) and Double border mode (B). Mode can be switched before calculation by the toggle button.